### DISCUSSION TITLE
All chakra styles are duplicated in initial render - outside of <html>

### DISCUSSION THREAD

**Discussion by Southclaws** [chakra-ui/chakra-ui#8803] [General]
### Description

I noticed this while trying to get my global.css to override chakra's reset (probably another issue, but probably Panda/Emotion related) and noticed all Chakra styles are sent to the client _twice_. Once inside the `<head>` in a style tag `<style data-emotion="css 1b7scut 7vw66x">` and again _outside_ `<html>` in a style tag `<style data-emotion="css 1b7scut 7vw66x">`

I managed to reproduce this one by just setting up a blank Next.js 13 app (with app directory) and setting up Chakra.

It might be Emotion but it's hard to tell. If I remove `CacheProvider` from the root then the problem goes away.

I've also tried `<ChakraProvider resetCSS={false} disableGlobalStyle={true}>`  but this doesn't change anything.

### Link to Reproduction

https://github.com/Southclaws/chakra-duplicate-styles

### Steps to reproduce

Run locally, `curl` or `http` get `http://localhost:3000`

Observe every style is duplicated:

<img width="1680" alt="image" src="https://github.com/chakra-ui/chakra-ui/assets/1636971/4793245b-9d50-4273-909a-48610abd0514">

Here's a HTML paste for ref

### Chakra UI Version

2.8.1

### Browser

Irrelevant

### Operating System

- [X] macOS
- [X] Windows
- [ ] Linux

**Comment by gioxoay** [comment]
Here is my implementation for CacheProvider to resolve this issue, the difference is that adding a logic check server already inserted style will ignore the client.

```

import { PropsWithChildren, createElement, useMemo, useRef } from "react";
import { CacheProvider as EmotionCacheProvider } from "@emotion/react";
import emotion_createCache, { Options as CacheOptions } from "@emotion/cache";
import { useServerInsertedHTML } from "next/navigation";

type EmotionCacheOptions = Partial<CacheOptions>;

type CacheProviderProps = PropsWithChildren<EmotionCacheOptions>;

const createCache = ((emotion_createCache as any).default ??
  emotion_createCache) as typeof emotion_createCache;

function useEmotionCache(options?: EmotionCacheOptions) {
  const isServerInserted = useRef(false);

  const cache = useMemo(() => {
    const cache = createCache({ key: "css", prepend: true, ...options });
    cache.compat = true;
    return cache;
  }, [options]);

  useServerInsertedHTML(() => {
    if (!isServerInserted.current) {
      isServerInserted.current = true;
      return createElement("style", {
        key: cache.key,
        "data-emotion": `${cache.key} ${Object.keys(cache.inserted).join(" ")}`,
        dangerouslySetInnerHTML: {
          __html: Object.values(cache.inserted).join(" "),
        },
      });
    }
  });

  return cache;
}

export function CacheProvider({
  children,
  ...cacheOptions
}: CacheProviderProps) {
  return (
    <EmotionCacheProvider value={useEmotionCache(cacheOptions)}>
      {children}
    </EmotionCacheProvider>
  );
}```

**Comment by gnowland** [comment]
The maintainers should consider merging this!

I can't find the source code for [@chakra-ui/next-js](https://www.npmjs.com/package/@chakra-ui/next-js) where this is implemented, otherwise I'd submit a PR 🥲. It used to be at [chakra-ui/packages/integrations/next-js/src/use-emotion-cache.ts](https://github.com/chakra-ui/chakra-ui/blob/38acfe89c5d1f1edc67bbc44e2edd38980ca3e08/packages/integrations/next-js/src/use-emotion-cache.ts) but it's apparently been moved...

@segunadebayo can you please point me to it?


Duplicates #7893

**Comment by segunadebayo** [comment]
@gnowland  Here's the reference to the v2 branch where you can implement this
https://github.com/chakra-ui/chakra-ui/tree/v2/packages
